scripts/markdown/check-markdown.js

```diff
@@ -134,13 +126,33 @@ function checkMarkdownFiles(folderPath, validateFunction, relativePath = '') {
                         process.exitCode = 1;
                     }
                 }
-            });
         });
-    });
+
+        await Promise.all(filePromises);
+    } catch (err) {
+        console.error(`Error in directory ${folderPath}:`, err);
+        throw err;
+    }
 }
 
 const docsFolderPath = path.resolve(__dirname, '../../markdown/docs');
 const blogsFolderPath = path.resolve(__dirname, '../../markdown/blog');
 
-checkMarkdownFiles(docsFolderPath, validateDocs);
-checkMarkdownFiles(blogsFolderPath, validateBlogs);
+async function main() {
+    try {
+        await Promise.all([
+            checkMarkdownFiles(docsFolderPath, validateDocs),
+            checkMarkdownFiles(blogsFolderPath, validateBlogs)
+        ]);
+    } catch (error) {
+        console.error('Failed to validate markdown files:', error);
+        process.exit(1);
+    }
+}
+
+/* istanbul ignore next */
```

Comment:
Why this comment is added?

Reply:
It is added to exclude the if block beneath it from coverage reporting. This block ensures that the script runs its main() function only when executed directly (and not when imported as a module). Without this check, the main() function would execute automatically when the script is imported during testing.